src/pal/src/include/pal/virtual.h

```diff
@@ -57,12 +57,15 @@ enum VIRTUAL_CONSTANTS
     VIRTUAL_NOACCESS,
     VIRTUAL_EXECUTE,
     VIRTUAL_EXECUTE_READ,
-    
-    VIRTUAL_PAGE_SIZE       = 0x1000,
-    VIRTUAL_PAGE_MASK       = VIRTUAL_PAGE_SIZE - 1,
+
+    VIRTUAL_PAGE_SIZE_MAX = 0x10000,
+
     BOUNDARY_64K    = 0xffff
 };
 
+#define VIRTUAL_PAGE_SIZE getpagesize()
```

Comment:
I would also replace the VIRTUAL_PAGE_SIZE macro usages with a function call (e.g. GetVirtualPageSize) and cache the getpagesize call.
All of the VIRTUAL_PAGE_MASK usages are for alignment purposes and using ALIGN_UP / ALIGN_DOWN / IS_ALIGNED with the virtual page size as the parameter would eliminate the need for this macro.